stake-pool/cli/src/main.rs

```diff
@@ -562,6 +562,104 @@ fn command_withdraw(
     Ok(None)
 }
 
+fn command_set_staking_auth(
+    config: &Config,
+    pool: &Pubkey,
+    stake_account: &Pubkey,
+    new_staker: &Pubkey,
+) -> CommandResult {
+    let pool_data = config.rpc_client.get_account_data(&pool)?;
+    let pool_data: StakePool = PoolState::deserialize(pool_data.as_slice())
+        .unwrap()
+        .stake_pool()
+        .unwrap();
+
+    let pool_withdraw_authority: Pubkey = PoolProcessor::authority_id(
+        &spl_stake_pool::id(),
+        pool,
+        PoolProcessor::AUTHORITY_WITHDRAW,
+        pool_data.withdraw_bump_seed,
+    )
+    .unwrap();
+
+    let mut transaction = Transaction::new_with_payer(
+        &[set_staking_authority(
+            &spl_stake_pool::id(),
+            &pool,
+            &config.owner.pubkey(),
+            &pool_withdraw_authority,
+            &stake_account,
+            &new_staker,
+            &stake_program_id(),
+        )?],
+        Some(&config.fee_payer.pubkey()),
+    );
+
+    let (recent_blockhash, fee_calculator) = config.rpc_client.get_recent_blockhash()?;
+    check_fee_payer_balance(config, fee_calculator.calculate_fee(&transaction.message()))?;
+    let mut signers = vec![config.fee_payer.as_ref(), config.owner.as_ref()];
+    unique_signers!(signers);
+    transaction.sign(&signers, recent_blockhash);
+    Ok(Some(transaction))
+}
+
+fn command_set_owner(
+    config: &Config,
+    pool: &Pubkey,
+    new_owner: &Option<Pubkey>,
+    new_fee_receiver: &Option<Pubkey>,
+) -> CommandResult {
+    let pool_data = config.rpc_client.get_account_data(&pool)?;
+    let pool_data: StakePool = PoolState::deserialize(pool_data.as_slice())
+        .unwrap()
+        .stake_pool()
+        .unwrap();
+
+    // If new accounts are missing in the arguments use the old ones
+    let new_owner: Pubkey = match new_owner {
+        None => pool_data.owner,
+        Some(value) => *value,
+    };
+    let new_fee_receiver: Pubkey = match new_fee_receiver {
+        None => pool_data.owner,
+        Some(value) => {
+            // Check for fee receiver being a valid token account and have to same mint as the stake pool
+            let account_data = config.rpc_client.get_account_data(value)?;
+            let account_data: TokenAccount =
+                match TokenAccount::unpack_from_slice(account_data.as_slice()) {
+                    Ok(data) => data,
+                    Err(_) => {
+                        println!("{} is not a token account", value);
+                        return Ok(None);
+                    }
+                };
+            if account_data.mint != pool_data.pool_mint {
+                println!("Fee receiver account belongs to a different mint");
+                return Ok(None);
```

Comment:
Instead of doing a `println!` and `return Ok(None)` we'd be better off returning an `Err` with the message, and then catching / printing the error at the processor level

Reply:
Yes, this was bothering me as well